api/internal/builtins/SortOrderTransformer.go

```diff
@@ -118,11 +109,13 @@ type legacyIDSorter struct {
 	// resids only stores the metadata of the object. This is an optimization as
 	// it's expensive to compute these again and again during ordering.
 	resids     []resid.ResId
+	resources []*resource.Resource
```

Comment:
I think it's worth adding a note here to explain why we do things this way and why it matters.
Basically, what happens is this:
- Initially, we sorted the metadata (ResId) of each object and then called `GetByCurrentId` on each to construct the  final list.
- The problem is that `GetByCurrentId` is inefficient and does a linear scan in a list every time we do that.
- So instead, we sort resources alongside the ResIds.

@natasha41575 I wonder if `GetByCurrentId` could use some better data structure in the future.